File: beecrowd/1040 - Média 3.rs
use std::io;

fn main() {

    let mut entrada: String = String::new();

    io::stdin()
        .read_line(&mut entrada)
        .expect("Falha ao ler uma linha");

    let valores: Vec<f64> = entrada.trim()
                    .split(" ")
                    .map(|x| x.trim().parse::<f64>().expect("Falha ao interpretar como ponto flutuante"))
                    .collect();

    let n1 = valores[0];
    let n2 = valores[1];
    let n3 = valores[2];
    let n4 = valores[3];

    let media = (2.0 * n1 + 3.0 * n2 + 4.0 * n3 + n4) / 10.0;

    println!("Media: {:.1}", media);

    if media >= 7.0 {
        println!("Aluno aprovado.");
    } else if media >= 5.0 {
        println!("Aluno em exame.");

        entrada = String::new();

        io::stdin()
            .read_line(&mut entrada)
            .expect("Falha ao ler uma linha");

        let nota_exame = entrada.trim()
                            .parse::<f64>()
                            .expect("Falha ao interpretar como ponto flutuante");

        println!("Nota do exame: {:.1}", nota_exame);

        let media_final = (media + nota_exame) / 2.0;

        if media_final >= 5.0 {
            println!("Aluno aprovado.");
        } else {
            println!("Aluno reprovado.");
        }

        println!("Media final: {:.1}", media_final);

    } else {
        println!("Aluno reprovado.");
    }
}

File: beecrowd/1040 - Média 3.go
package main

import (
	"fmt"
)

func main() {
	var n1, n2, n3, n4 float64

	fmt.Scan(&n1, &n2, &n3, &n4)

	media := (2.0*n1 + 3.0*n2 + 4.0*n3 + n4) / 10.0

	fmt.Printf("Media: %.1f\n", media)

	if media < 5.0 {
		fmt.Println("Aluno reprovado.")
	} else if media < 7.0 {
		var n5 float64

		fmt.Println("Aluno em exame.")
		fmt.Scan(&n5)
		fmt.Printf("Nota do exame: %.1f\n", n5)

		media_final := (media + n5) / 2.0

		if media_final < 5.0 {
			fmt.Println("Aluno reprovado.")
		} else {
			fmt.Println("Aluno aprovado.")
		}

		fmt.Printf("Media final: %.1f\n", media_final)
	} else {
		fmt.Println("Aluno aprovado.")
	}
}
